src/Microsoft.SqlTools.ResourceProvider.Core/Common/ConcurrentCache.cs

```diff
@@ -0,0 +1,82 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.SqlTools.ResourceProvider.Core
+{
+    internal class ConcurrentCache<T>
```

Comment:
>ConcurrentCache [](start = 19, length = 15)

some of the "utility" classes look like they may be interesting in other use cases.  it may be worthwhile to stick them in hosting or another shared library so they've readily available.

Reply:
Yeah, will do